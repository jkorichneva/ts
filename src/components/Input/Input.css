.input {
    appearance: none;
    -webkit-appearance: none;
    border-radius: 5px;
    border: 2px solid #ccc;
    padding: 5px 10px;
    box-sizing: border-box;
    outline: none;
    display: block;
    margin-top: 20px;
}

.input:hover,
.input:focus {
    border-color: #999;
}

.input_size-s {
    height: 40px;
    font-size: 16px;
}

.input_size-m {
    height: 50px;
    font-size: 21px;
}

.input_size-l {
    height: 60px;
    font-size: 26px;
}

.input_dark {
    background: transparent;
    color: white;
    border-color: white;
}

.input_dark::placeholder {
    color: #ccc;
}
