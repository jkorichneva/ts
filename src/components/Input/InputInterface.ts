import React,{ ChangeEvent } from 'react';

export default interface InputInterface {
    value?: string|number;
    placeholder: string;
    type: string;
    theme?: string;
    size?: string;
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
