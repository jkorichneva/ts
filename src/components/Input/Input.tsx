import React, {FunctionComponent} from 'react';
import InputInterface from './InputInterface';

import './Input.css';

export const Input: FunctionComponent<InputInterface> =
    ({  value,
        placeholder,
        type,
        theme = 'light',
        size= 'm',
        onChange
    }) => {
        // tslint:disable-next-line:max-line-length
        const onChangeInput: (event: React.ChangeEvent<HTMLInputElement>) => void = (event: React.ChangeEvent<HTMLInputElement>) => {
            event.persist();
            onChange(event);
        };
        return (
            <input
                className={`input input_${theme} input_size-${size}`}
                type={type}
                value={value}
                placeholder={placeholder}
                onChange={onChangeInput}
            />
        );
    }
export default Input;
