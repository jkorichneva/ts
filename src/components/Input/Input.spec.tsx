import React from 'react';
import { mount } from 'enzyme';
import Input from './Input';

test('renders Input', () => {
    const spy = jest.fn();
    const wrapper = mount(<Input  onChange={spy} placeholder="Инпут" type="text" />);
    expect(wrapper).toMatchSnapshot();
});

test('should change Input', () => {
    const spy = jest.fn();
    const wrapper = mount(<Input  onChange={spy} placeholder="Инпут" type="text" />);
    const input = wrapper.find('input');
    input.simulate('change');
    expect(spy).toHaveBeenCalled();
});
