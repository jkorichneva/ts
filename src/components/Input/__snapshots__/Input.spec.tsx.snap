// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input 1`] = `ReactWrapper {}`;
